libpages/server.go

```diff
@@ -33,20 +33,38 @@ type ServerConfig struct {
 	UseStaging       bool
 	Logger           *zap.Logger
 	UseDiskCertCache bool
+	StatsReporter    StatsReporter
+
+	whiteList     map[string]bool
+	whiteListOnce sync.Once
+}
+
+func (c *ServerConfig) allowedByDomainWhiteList(domain string) bool {
+	c.whiteListOnce.Do(func() {
+		if len(c.DomainWhitelist) > 0 {
+			c.whiteList = make(map[string]bool, len(c.DomainWhitelist))
```

Comment:
Now that `whitelist` is in this config, why is this separation from `DomainWhitelist` necessary?

Reply:
The DomainWhitelist is a slice, while whitelist is a map.